tests/unit/javascript/conftest.py

```diff
@@ -134,7 +191,78 @@ def run(self, source):
         return self.webview.page().mainFrame().evaluateJavaScript(source)
 
 
+class JSWebEngineTester(JSTester):
+
+    """Object returned by js_tester_webengine which provides a webview.
+
+    Attributes:
+        webview: The webview which is used.
+        _qtbot: The QtBot fixture from pytest-qt.
+        _jinja_env: The jinja2 environment used to get templates.
+    """
+
+    def __init__(self, webview, callback_checker, qtbot):
+        JSTester.__init__(self, webview, qtbot)
+        self.webview.setPage(TestWebEnginePage(self.webview))
+        self.callback_checker = callback_checker
+
+    def load_file(self, path: str, force=False):
+        """Load a file from disk.
+
+        Args:
+            path: The string path from disk to load (relative to this file)
+            force: Whether to force loading even if the file is invalid.
+        """
+        self.load_url(QUrl.fromLocalFile(
+            os.path.join(os.path.dirname(__file__), path)), force)
+
+    def load_url(self, url: QUrl, force: bool = False):
+        """Load a given QUrl.
+
+        Args:
+            url: The QUrl to load.
+            force: Whether to force loading even if the file is invalid.
+        """
+        with self._qtbot.waitSignal(self.webview.loadFinished) as blocker:
+            self.webview.load(url)
+        if not force:
+            assert blocker.args == [True]
+
+    def run_file(self, filename: str, expected) -> None:
+        """Run a javascript file.
+
+        Args:
+            filename: The javascript filename, relative to
+                      qutebrowser/javascript.
+            expected: The value expected return from the javascript execution
+        """
+        source = utils.read_file(os.path.join('javascript', filename))
+        self.run(source, expected)
+
+    def run(self, source: str, expected,
+            world=QWebEngineScript.ApplicationWorld) -> None:
```

Comment:
This [fails](https://travis-ci.org/qutebrowser/qutebrowser/jobs/300287911) without QtWebEngine being available - you'll probably need to set this to `None` or so and handle it in the body.